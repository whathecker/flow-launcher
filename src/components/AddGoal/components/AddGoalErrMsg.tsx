import React from "react";
import { StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Typography } from "../../../styles";

type AddGoalErrMsgProps = {
  msg: string;
};

const AddGoalErrMsg: React.FC<AddGoalErrMsgProps> = ({
  msg,
}: AddGoalErrMsgProps) => {
  return (
    <View style={styles.errorMsgWrapper}>
      <Text
        lightColor="#c84827"
        darkColor="#c84827"
        style={styles.errorMsgText}
      >
        {msg}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  errorMsgWrapper: {
    paddingTop: 5,
    paddingBottom: 3,
    paddingLeft: 5,
  },
  errorMsgText: {
    ...Typography.p,
    fontSize: 16,
  },
});

export default AddGoalErrMsg;
