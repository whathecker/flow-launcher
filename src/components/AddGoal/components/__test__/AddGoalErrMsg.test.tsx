import * as React from "react";
import renderer from "react-test-renderer";
import AddGoalErrMsg from "../AddGoalErrMsg";

describe("Test AddGoalErrMsg component", () => {
  it("Should renders AddGoalFormLabel correctly", () => {
    const tree = renderer
      .create(<AddGoalErrMsg msg="Error Message" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
