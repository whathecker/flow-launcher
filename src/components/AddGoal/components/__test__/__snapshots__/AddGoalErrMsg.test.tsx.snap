// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddGoalErrMsg component Should renders AddGoalFormLabel correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "paddingBottom": 3,
        "paddingLeft": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#c84827",
        },
        Object {
          "fontFamily": "Lexend-Regular",
          "fontSize": 16,
        },
      ]
    }
  >
    Error Message
  </Text>
</View>
`;
